app/assets/v2/js/grants/cart/algorand_extension.js

```diff
@@ -1,207 +1,765 @@
+        }
+
+        // step4: set modal to waiting state
+        vm.updatePaymentStatus(grant.grant_id, "waiting");
+
+        // step5: get txnParams
+        const algodClient = new algosdk.Algodv2(
+          "",
+          "https://node.algoexplorerapi.io",
+          ""
+        );
+        const params = await algodClient.getTransactionParams().do();
+        const enc = new TextEncoder(); // always utf-8
+
+        let txn;
+        // step5: sign transaction
 
-      }).catch(e => {
+        if (token_name == "ALGO") {
+          // ALGO token
+          txn = {
+            suggestedParams: {
+              ...params,
+            },
+            from: from_address.toUpperCase(),
+            to: to_address.toUpperCase(),
+            amount: amount * 10 ** 6,
+            note: enc.encode("contributing to gitcoin grant"),
+            type: "pay",
+          };
+        } else {
+          // ALGO assets
+          txn = {
+            from: from_address.toUpperCase(),
+            to: to_address.toUpperCase(),
+            assetIndex: Number(token.addr),
+            note: enc.encode("contributing to gitcoin grant"),
+            amount: amount * 10 ** token.decimals,
+            type: "axfer",
+            suggestedParams: {
+              ...params,
+            },
+          };
+        }
+        let binaryTx = ((tx) => {
+          tx.flatfee = true;
+          tx.fee = 1000;
+          return tx;
+        })(new algosdk.Transaction(txn)).toByte();
+        myAlgoConnect
+          .signTransaction(binaryTx)
+          .then((stx) => {
+            // step7: broadcast txn
+            algodClient
+              .sendRawTransaction(stx.blob)
+              .do()
+              .then((tx) => {
+                contributeWithAlgorandExtensionCallback(
+                  null,
+                  tx.txId,
+                  grant,
+                  vm,
+                  from_address
+                );
+              })
+              .catch((e) => {
+                console.log(e);
+                _alert(
+                  {
+                    message:
+                      "Unable to broadcast transaction. Please try again",
+                  },
+                  "danger"
+                );
+                vm.updatePaymentStatus(grant.grant_id, "failed");
+                return;
+              });
+          })
+          .catch((e) => {
+            console.log(e);
+            _alert(
+              { message: "Unable to sign transaction. Please try again" },
+              "danger"
+            );
+            vm.updatePaymentStatus(grant.grant_id, "failed");
+            return;
+          });
+      })
+      .catch((e) => {
         console.log(e);
-        _alert({ message: 'Unable to sign transaction. Please try again' }, 'danger');
-        vm.updatePaymentStatus(grant.grant_id, 'failed');
+        _alert(
+          {
+            message: "Please allow Gitcoin to connect to AlgoSigner extension",
```

Comment:
this error message should reflect a failed MyAlgo wallet transaction, not AlgoSigner

Reply:
Fixed